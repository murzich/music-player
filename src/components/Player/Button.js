import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const propTypes = {
  callback: PropTypes.func.isRequired,
  extraClass: PropTypes.string,
  children: PropTypes.node,
};
const defaultProps = {
  extraClass: '',
  children: 'button',
};

function Button({
  callback,
  extraClass,
  children,
}) {
  const className = `Button ${extraClass}`;

  return (
    <button
      onClick={callback}
      className={className}
    >
      {children}
    </button>
  );
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
