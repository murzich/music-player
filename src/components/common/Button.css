.Button {
  margin: 5px 0;
  padding: .5rem;
  color: var(--color-primary);
  border: none;
  border-radius: 2px;
  background: var(--bg-main);
  transition: all 200ms ease;
  font-size: 1rem;
  font-weight: 700;
}
.Button:hover {
  background: var(--bg-body);
}
