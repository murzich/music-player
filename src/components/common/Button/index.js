import React from 'react';
import PropTypes from 'prop-types';

import style from './Button.css';

function Button(props) {
  const {
    children,
    type,
    className,
    ...restProps
  } = props;
  return (
    <button type={type} className={className} {...restProps}>
      { children }
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node,
  type: PropTypes.string,
  className: PropTypes.string,
};
Button.defaultProps = {
  children: 'default button',
  type: 'button',
  className: style.Button,
};

export default Button;

