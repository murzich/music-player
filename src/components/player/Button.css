.Button {
  background: none;
  border: none;
  color: var(--controls-secondary);
  font-size: 1.25rem;
  padding: 0;
  margin: 0;
  transition: all 200ms ease;
}
.Button:hover {
  color: var(--color-primary);
}
.Button:focus {
  outline: 0;
}
.Button-main {
  composes: Button;
  margin: 0 1.5rem;
  font-size: 2rem;
  color: var(--controls-primary);
}
