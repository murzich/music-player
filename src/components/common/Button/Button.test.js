/* eslint-disable function-paren-newline */
import React from 'react';
import * as renderer from 'react-test-renderer';
import Button from './index';

describe('Button component', () => {
  test('should render default with values', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.type).toBe('button');
    expect(tree.children).toContain('default button');
    expect(tree.props.type).toBe('button');
  //  TODO: test for default className;
  });

  test('should render with given props', () => {
    const element = (
      <Button
        type="submit"
        className="test__className"
        otherProps="otherProps"
        anotherProps={[1, 2, 3]}
      >
        childrenText
      </Button>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.children).toContain('childrenText');
    expect(tree.props.type).toBe('submit');
    expect(tree.props.className).toBe('test__className');
    expect(tree.props.otherProps).toBeTruthy();
    expect(tree.props.otherProps).toBe('otherProps');
    expect(tree.props.anotherProps).toEqual([1, 2, 3]);
  });
});

