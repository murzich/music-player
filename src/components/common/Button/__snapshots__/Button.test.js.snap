// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render default with values 1`] = `
<button
  className={undefined}
  type="button"
>
  default button
</button>
`;

exports[`Button component should render with given props 1`] = `
<button
  anotherProps={
    Array [
      1,
      2,
      3,
    ]
  }
  className="test__className"
  otherProps="otherProps"
  type="submit"
>
  childrenText
</button>
`;
